docs/tools.md

```diff
@@ -0,0 +1,117 @@
+# Tools
+
+## Introduction
+
+Tools in the context of an agent refer to additional functionalities or capabilities integrated with the agent to perform specific tasks beyond text processing.
+
+These tools extend the agent's abilities, allowing it to interact with external systems, access information, and execute actions.
+
+## Development
+
+### Writing a new tool
+
+To create a tool, the `Tool` class must be extended/ implemented or `DynamicTool` created from [tools base](../src/tools/base.ts). When starting to write tools, it is recommended to implement the `Tool` class. The `DynamicTool` class is an extension of the `Tool` class that allows for dynamic forms of input. Refer to the list of [examples](#examples) for some simple examples or any of the built-in tools in order to guide the creation of new tools.
+
+Tools MUST do the following:
+
+- Implement the `Tool` class:
+
+  `MyNewToolOutput` is required, must be an implementation of `ToolOutput` such as `StringToolOutput` or `JSONToolOutput`
+
+  `ToolOptions` is optional (default BaseToolOptions), constructor parameters that are passed during tool creation
+
+  `ToolRunOptions` is optional (default BaseToolRunOptions), optional parameters that are passed to the run method
+
+  ```typescript
+  export class MyNewTool extends Tool<MyNewToolOutput, ToolOptions, ToolRunOptions> {
```

Comment:
show import path

Reply:
Resolved (pending further review) in 3863b50

Reply:
Remove the JS extension - it will not work.
Add some new lines.